web-app/backend/apps/common/utils.py

```diff
@@ -0,0 +1,90 @@
+import os
+
+from kubeflow.kubeflow.crud_backend import api, helpers, logging
+
+log = logging.getLogger(__name__)
+
+FILE_ABS_PATH = os.path.abspath(os.path.dirname(__file__))
+
+INFERENCESERVICE_TEMPLATE_YAML = os.path.join(
+    FILE_ABS_PATH, "yaml", "inference_service_template.yaml")
+
+
+def load_inference_service_template(**kwargs):
+    """
+    kwargs: the parameters to be replaced in the yaml
+
+    Reads the yaml for the web app's custom resource, replaces the variables
+    and returns it as a python dict.
+    """
+    return helpers.load_param_yaml(INFERENCESERVICE_TEMPLATE_YAML, **kwargs)
+
+
+# helper functions for accessing the logs of an InferenceService
+def get_inference_service_pods(svc, components=[]):
+    """
+    Return a dictionary with (endpoint, component) keys,
+    i.e. ("default", "predictor") and a list of pod names as values
+    """
+    namespace = svc["metadata"]["namespace"]
+
+    # dictionary{revisionName: (endpoint, component)}
+    revisions_dict = get_components_revisions_dict(components, svc)
+
+    if len(revisions_dict.keys()) == 0:
+        return {}
+
+    pods = api.list_pods(namespace, auth=False).items
+    component_pods_dict = {}
+    for pod in pods:
+        for revision in revisions_dict:
+            if pod.metadata.generate_name is None:
+                continue
+
+            if not pod.metadata.generate_name.startswith(revision):
+                continue
+
+            component = revisions_dict[revision]
+            curr_pod_names = component_pods_dict.get(component, [])
+            curr_pod_names.append(pod.metadata.name)
+            component_pods_dict[component] = curr_pod_names
```

Comment:
That's a way better approach from parsing names!
I'll add a commit to address this

Reply:
@yuzisun I pushed a commit that uses the `serving.knative.dev/revision` label to filter which Pods belong to a specific Revision. I also tested it and ensure everything is working as expected.

I also added a [small section](https://docs.google.com/document/d/1MaCqHeqd7NiP1nseWkzqqAntRr3PiEH7k_7WnKjQSPU/edit#heading=h.rm3tua9urp81) in the design doc to keep track of this decision.

I believe this can be resolved so I'll resolve the convo, but feel free to add any further comments if you'd like some more changes